pkg/sink/kafka/admin.go

```diff
@@ -39,11 +52,44 @@ func NewSaramaAdminClient(ctx context.Context, config *Options) (ClusterAdminCli
 	if err != nil {
 		return nil, err
 	}
-	return &saramaAdminClient{client: client}, nil
+	return &saramaAdminClient{
+		client:          client,
+		brokerEndpoints: config.BrokerEndpoints,
+		config:          saramaConfig,
+	}, nil
+}
+
+func (a *saramaAdminClient) reset() error {
+	newClient, err := sarama.NewClusterAdmin(a.brokerEndpoints, a.config)
+	if err != nil {
+		return cerror.Trace(err)
+	}
+
+	_ = a.client.Close()
+	a.client = newClient
+
+	return errors.New("retry after reset")
```

Comment:
why do we always return an error?

Reply:
if return nil here, the retry logic will exit, this return an error is on purpose.

Reply:
Perhaps it is more appropriate to return this error outside of reset?

Reply:
it's very strange that a func always returns an error.

Reply:
if we move it outside, we still have to write one error when retry, no matter reset return error or not.